src/main/java/net/dv8tion/jda/api/interactions/components/text/TextInput.java

```diff
@@ -0,0 +1,488 @@
+         *
+         * @param  required 
+         *         If this TextInput should be required
+         *
+         * @return The same builder instance for chaining
+         */
+        @Nonnull
+        public Builder setRequired(boolean required)
+        {
+            this.required = required;
+            return this;
+        }
+
+        /**
+         * Sets the minimum length of this input field. Default is -1 (No min length).
+         *
+         * <b>This has to be between 0 and {@value #TEXT_INPUT_MAX_LENGTH}</b>
+         *
+         * @param  minLength
+         *         The minimum amount of characters that need to be written
+         *
+         * @throws IllegalArgumentException
+         *         If minLength is negative or greater than {@value #TEXT_INPUT_MAX_LENGTH}
+         *
+         * @return The same builder instance for chaining
+         */
+        @Nonnull
+        public Builder setMinLength(int minLength)
+        {
+            Checks.notNegative(minLength, "Minimum length");
+            Checks.check(minLength <= TEXT_INPUT_MAX_LENGTH, "Minimum length cannot be longer than %d characters!", TEXT_INPUT_MAX_LENGTH);
+            this.minLength = minLength;
+            return this;
+        }
+
+        /**
+         * Sets the maximum length of this input field. Default is -1 (No max length).
+         *
+         * <p><b>This has to be between 1 and {@value #TEXT_INPUT_MAX_LENGTH}</b>
+         *
+         * @param  maxLength 
+         *         The maximum amount of characters that need to be written
+         *
+         * @throws IllegalArgumentException
+         *         If maxLength is smaller than 1 or greater than {@value #TEXT_INPUT_MAX_LENGTH}
+         *
+         * @return The same builder instance for chaining
+         */
+        @Nonnull
+        public Builder setMaxLength(int maxLength)
+        {
+            Checks.check(maxLength >= 1, "Maximum length cannot be smaller than 1 character!");
+            Checks.check(maxLength <= TEXT_INPUT_MAX_LENGTH, "Maximum length cannot be longer than " + TEXT_INPUT_MAX_LENGTH + " characters!");
+
+            this.maxLength = maxLength;
+            return this;
+        }
+
+        /**
+         * Sets the minimum and maximum required length on this TextInput component
+         *
+         * @param  min 
+         *         Minimum length of the text input
+         * @param  max 
+         *         Maximum length of the text input
+
+         * @throws IllegalArgumentException
+         *         <ul>
+         *             <li>If min is negative or greater than {@link #TEXT_INPUT_MAX_LENGTH}</li>
+         *             <li>If max is smaller than 1, smaller than min or greater than {@link #TEXT_INPUT_MAX_LENGTH}</li>
+         *         </ul>
+         *
+         * @return The same builder instance for chaining
+         */
+        @Nonnull
+        public Builder setRequiredRange(int min, int max)
```

Comment:
Shouldn't this also set required to true?

Reply:
No, since it only sets the required bounds when the user provides data

Reply:
Might be better named `setLengthRange` or something similar?

Reply:
That's also how it's called on `OptionData`